### PR TITLE
Update libp2p 0.17.0 -> 0.23.4 and all other deps

### PR DESCRIPTION
**Pull request by GreenFatGuy**
Update libp2p-core and deps

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     - name: Install Go
       uses: actions/setup-go@v2
       with:
-        go-version: 1.16.3
+        go-version: 1.19.3
```
Comment (justheuristic) [inline .github/workflows/release.yml:20]: Based on (limited) CI observations, it appears that 1.18 stuff either always works or is less likely to fail. If true, consider running with 1.18 for now

Comment (GreenFatGuy) [reply .github/workflows/release.yml:20]: Ok

Comment (GreenFatGuy) [reply .github/workflows/release.yml:20]: Thus, if we restart CI enough times - we will get all green

File: p2pd/main.go [hunk 2]
```diff
@@ -85,7 +84,7 @@ func main() {
 	relayEnabled := flag.Bool("relay", true, "Enables circuit relay")
 	flag.Bool("relayActive", false, "Enables active mode for relay (deprecated, has no effect)")
 	flag.Bool("relayHop", false, "Enables hop for relay (deprecated, has no effect)")
-	relayHopLimit := flag.Int("relayHopLimit", 0, "Sets the hop limit for hop relays")
+	relayHopLimit := flag.Int("relayHopLimit", 0, "DEPRICATED FEATURE, HAS NO EFFECT. Sets the hop limit for hop relays")
```
Comment (borzunov) [inline p2pd/main.go:87]:
```suggestion
	relayHopLimit := flag.Int("relayHopLimit", 0, "Sets the hop limit for hop relays (deprecated, has no effect)")
```

Please use the same format as for previous deprecated features.

File: .github/workflows/release.yml (hunk 1 above)
Comment (borzunov) [inline .github/workflows/release.yml:20]:
```suggestion
        go-version: 1.18.8
```

**Review by borzunov — APPROVED** [review]
As agreed with @justheuristic, we merge this despite the blinking tests because (a) hivemind tests don't seem to fail and (b) we need libp2p relay features in Petals.

**Pull request merged**